api/handlers/reporthandler.py

```diff
@@ -92,6 +63,38 @@ def build(self):
         """
         raise NotImplementedError()
 
+    @staticmethod
+    def _get_result_list(output):
+        """
+        Helper function for extracting mongo aggregation results
+
+        Given the output of a mongo aggregation call, checks 'ok' field
+        If not 1.0, 'result' field does not exist or 'result' array is empty,
+        throws APIReportException
+        """
+
+        if output.get('ok') == 1.0:
+            result = output.get('result')
+            if result is not None and len(result) > 0:
+                return result
```

Comment:
I know this functionality has been duplicated at least once, if you want consider deduping?